src/utils/server-utils.ts

```diff
@@ -302,15 +338,49 @@ export async function getLayersAvailableDates(): Promise<AvailableDates> {
     (layer): layer is PointDataLayerProps => layer.type === 'point_data',
   );
 
-  const layerDates: AvailableDates[] = await Promise.all([
+  const layerDates = await Promise.all([
     ...wmsServerUrls.map(url => getWMSCapabilities(url)),
     ...wcsServerUrls.map(url => getWCSCoverage(url)),
     ...pointDataLayers.map(async layer => ({
       [layer.id]: await getPointDataCoverage(layer),
     })),
   ]);
 
-  return merge({}, ...layerDates);
+  const mergedLayers: { [key: string]: number[] } = merge({}, ...layerDates);
```

Comment:
Let's add some comments in this part of the code as well

Reply:
Thinking about it more, the structure might be simpler if we do something like this:

const mergedLayers: { [key: string]: number[] } = merge({}, ...layerDates.map(layerDate -> expandDatesOverValidityPeriod(layerDate));

Then expandDatesOverValidityPeriod finds the corresponding LayerDefinition, if validity is a number -> it adds new dates. And for all layers, it transforms dates as number[] into DateItem[]

I believe this would make the code simpler and easier to expand upon once we add this feature for other layers.